code/code/cmd/cmd_deathstroke.cc

```diff
@@ -5,222 +5,168 @@
 
 static int deathstroke(TBeing *caster, TBeing *victim)
 {
-  int percent;
-  int i = 0;
   bool wasSuccess = FALSE; 
-  const int DEATHSTROKE_MOVE = 25;
+  const int DEATHSTROKE_MOVE = 15;
   TBaseWeapon *tw;
+  TThing *ob;
   spellNumT sktype = SKILL_DEATHSTROKE;
+  affectedData aff1, aff2;
   
+  // Ensure player even knows the skill before continuing
+  if (!caster->doesKnowSkill(SKILL_DEATHSTROKE)) {
+    caster->sendTo("You know nothing about deathblows.\n\r");
+    return FALSE;
+  }
+
+  // Ensure this isn't a peaceful room
   if (caster->checkPeaceful("You feel too peaceful to contemplate violence.\n\r"))
     return FALSE;
-  
-  if (caster->getCombatMode() == ATTACK_BERSERK) {
-    caster->sendTo("You are berserking! You can't focus enough to deathstroke anyone!\n\r ");
+
+  // Check for lockout from recently using this skill
+  if (caster->affectedBySpell(SKILL_DEATHSTROKE)) {
+    caster->sendTo("You are still recovering from your last deathstroke and cannot use this ability again at this time.\n\r");
     return FALSE;
   }
 
+  // Make sure the player has enough vitality to use the skill
   if (caster->getMove() < DEATHSTROKE_MOVE) {
     caster->sendTo("You don't have the vitality to make the move!\n\r");
     return FALSE;
   }
+
+  // Prevent players from attacking themselves
   if (victim == caster) {
     caster->sendTo("Do you REALLY want to kill yourself?...\n\r");
     return FALSE;
   }
+
+  // Avoid players attacking un-harmable victims
   if (caster->noHarmCheck(victim))
```

Comment:
No message sent? Or is that in noHarmCheck?

Reply:
Updated to add a quick message.

Reply:
Just FYI, noHarmCheck already sends a message based on circumstances - that's why it didn't have one here initially.

https://github.com/sneezymud/sneezymud/blob/221a9c1abea6439a7f682b220cee8cdf904633c3/code/code/misc/offense.cc#L2066-L2112